mlflow/utils/rest_utils.py

```diff
@@ -14,26 +14,76 @@
 from mlflow.utils.string_utils import strip_suffix
 from mlflow.exceptions import MlflowException, RestException
 
-_REST_API_PATH_PREFIX = "/api/2.0"
 RESOURCE_DOES_NOT_EXIST = "RESOURCE_DOES_NOT_EXIST"
+_REST_API_PATH_PREFIX = "/api/2.0"
+_DEFAULT_HEADERS = {"User-Agent": "mlflow-python-client/%s" % __version__}
+# Response codes that generally indicate transient network failures and merit client retries,
+# based on guidance from cloud service providers
+# (https://docs.microsoft.com/en-us/azure/architecture/best-practices/retry-service-specific#general-rest-and-retry-guidelines)
```

Comment:
Do we want to use this set of codes as the general retry-able cases? Currently this is only used in cloud_storage_http_request(). For http_request, we use a subset of it, namely, 429 / 500 / 503.

In addition, 500 seems not really retry-able but we can discuss what is retry-able another day.

Reply:
I think it makes sense to expand the retry-able cases for non-cloud-storage requests. Thanks Jin!

Reply:
Done.